packages/client/src/exampleTypes/pivot.js

```diff
@@ -161,6 +109,72 @@ export let skipResetExpansionsFields = [
 ]
 
 export default {
+  init: (node, extend, snapshot) => {
+    if (_.isNil(node.hasResults))
+      Object.defineProperty(node, 'hasResults', {
```

Comment:
Instead of adding this during init, it probably makes more sense to define it as a first class thing types can specify and have the core call `hasResults` on init and when processing responses for nodes. Using a getter is interesting since we can in theory defer computation until it's accessed - but I think in practice, it'll be used by anything observing it. Unless we're memoizing and caching the results, I think it actually ends up less performant than just calling whenever we process responses (since the code will only run once instead of on every access).